### PR TITLE
Add GPU configuration UI

### PR DESCRIPTION
**Pull request by tkpd-hafizhan**
**What this PR does / why we need it**:
This PR We add GPU Configuration to UI

Demo:

https://github.com/caraml-dev/merlin/assets/8122852/3dd125bf-7487-4a06-996b-9bc9191c88a7

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Allow user to request for GPU via Merlin UI
```

**Checklist**

- [ ] Added unit test, integration, and/or e2e tests
- [x] Tested locally
- [ ] Updated documentation
- [ ] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

File: ui/src/providers/environments/context.js [hunk 1]
```diff
@@ -6,6 +6,8 @@ const EnvironmentsContext = React.createContext([]);
 export const EnvironmentsContextProvider = ({ children }) => {
   const [{ data: environments }] = useMerlinApi("/environments", {}, []);
 
+  console.log("___environments", environments);
```
Comment (ariefrahmansyah) [inline ui/src/providers/environments/context.js:9]: Please remove this

File: ui/src/pages/version/components/forms/components/ResourcesPanel.js [hunk 2]
```diff
@@ -79,6 +146,60 @@ export const ResourcesPanel = ({
...
+                      {
+                        value: "None",
+                        inputDisplay: "None",
+                      },
+                      ...Object.keys(gpus).map((display_name) => ({
+                        value: display_name,
+                        inputDisplay: display_name,
+                      })),
+                    ]}
+                    valueOfSelected={resourcesConfig.gpu_display_name || "None"}
```
Comment (ariefrahmansyah) [inline ui/src/pages/version/components/forms/components/ResourcesPanel.js:174]: One more: Please add hasDividers prop here

File: ui/src/pages/version/components/forms/components/ResourcesPanel.js [hunk 3]
```diff
@@ -79,6 +146,60 @@ export const ResourcesPanel = ({
...
+                    />
+                  }
+                  isInvalid={!!errors.gpu_request}
+                  error={errors.gpu_request}
+                  fullWidth
+                >
+                  <EuiSuperSelect
+                    onChange={onGPUValueChange}
+                    options={gpuValueOptions}
+                    valueOfSelected={resourcesConfig.gpu_request || "None"}
```
Comment (ariefrahmansyah) [inline ui/src/pages/version/components/forms/components/ResourcesPanel.js:194]: One more: Please add hasDividers prop here as well

**Review by ariefrahmansyah — APPROVED** [review]
Great works! Thanks, @tkpd-hafizhan!

File: ui/src/pages/version/components/forms/components/ResourcesPanel.js [hunk 4]
```diff
@@ -8,26 +14,87 @@ import {
...
 } from "@elastic/eui";
 import { FormLabelWithToolTip, useOnChangeHandler } from "@caraml-dev/ui-lib";
 import { Panel } from "./Panel";
 import { calculateCost } from "../../../../../utils/costEstimation";
+import EnvironmentsContext from "../../../../../providers/environments/context";
 
 const maxTicks = 20;
 
 export const ResourcesPanel = ({
+  environment,
```
Comment (krithika369) [inline ui/src/pages/version/components/forms/components/ResourcesPanel.js:27]: Late to the party but shall we control the display of GPU options with a separate prop like `isGPUEnabled` and set it to true when the `<ResourcesPanel>` is created from `ModelStep.js` but leave it unset for `TransformerStep.js`?

Right now, the `environment` value is controlling whether we show GPU options or not which is not very clear immediately.

Comment (ariefrahmansyah) [reply ui/src/pages/version/components/forms/components/ResourcesPanel.js:27]: Thanks for catching this, Krithika! We will update it accordingly! cc: @tkpd-hafizhan

**Pull request merged**